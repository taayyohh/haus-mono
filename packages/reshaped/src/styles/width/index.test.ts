import getWidthStyles from "./index";

describe("Styles/Width", () => {
	test("handles px value", () => {
		expect(getWidthStyles("50px")).toMatchSnapshot();
	});

	test("handles % value", () => {
		expect(getWidthStyles("50%")).toMatchSnapshot();
	});

	test("handles unit value", () => {
		expect(getWidthStyles(5)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(getWidthStyles()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(getWidthStyles({ s: "50px", l: "50%" })).toMatchSnapshot();
	});
});
