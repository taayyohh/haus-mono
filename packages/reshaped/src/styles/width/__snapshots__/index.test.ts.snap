// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles/Width > handles % value 1`] = `
{
  "classNames": [
    "_literal_dd7142",
  ],
  "variables": {
    "--rs-w-s": "50%",
  },
}
`;

exports[`Styles/Width > handles px value 1`] = `
{
  "classNames": [
    "_literal_dd7142",
  ],
  "variables": {
    "--rs-w-s": "50px",
  },
}
`;

exports[`Styles/Width > handles responsive value 1`] = `
{
  "classNames": [
    "_literal_dd7142",
    "_literal--l_dd7142",
  ],
  "variables": {
    "--rs-w-l": "50%",
    "--rs-w-s": "50px",
  },
}
`;

exports[`Styles/Width > handles undefined value 1`] = `null`;

exports[`Styles/Width > handles unit value 1`] = `
{
  "classNames": [
    "_unit_dd7142",
  ],
  "variables": {
    "--rs-w-s": 5,
  },
}
`;
